server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java

```diff
@@ -389,9 +410,38 @@ public boolean apply(@Nonnull String input) {
 
   private enum ComponentDtoWithSnapshotIdToCopyResourceIdFunction implements Function<ComponentDtoWithSnapshotId, Long> {
     INSTANCE;
+
     @Override
     public Long apply(@Nonnull ComponentDtoWithSnapshotId input) {
       return input.getCopyResourceId();
     }
   }
+
+  private static class HasMeasure implements Predicate<ComponentDtoWithSnapshotId> {
+    private final Table<String, MetricDto, MeasureDto> measuresByComponentUuidAndMetric;
+    private final MetricDto metric;
+
+    private HasMeasure(Table<String, MetricDto, MeasureDto> measuresByComponentUuidAndMetric, MetricDto metric) {
+      this.measuresByComponentUuidAndMetric = measuresByComponentUuidAndMetric;
+      this.metric = metric;
+    }
+
+    @Override
+    public boolean apply(@Nonnull ComponentDtoWithSnapshotId input) {
+      return measuresByComponentUuidAndMetric.get(input.uuid(), metric) != null;
+    }
+  }
+
+  private static class MatchMetricKey implements Predicate<MetricDto> {
+    private final String metricKeyToSort;
+
+    private MatchMetricKey(String metricKeyToSort) {
+      this.metricKeyToSort = metricKeyToSort;
+    }
+
+    @Override
+    public boolean apply(@Nonnull MetricDto input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)